src-data/witan/models/upload_data.clj

```diff
@@ -0,0 +1,31 @@
+(ns witan.models.upload-data
+  (:require [clojure.string :as str]
+            [amazonica.aws.s3 :as aws]
+            [amazonica.aws.s3transfer :as aws-s3])
+  (:gen-class))
+
+(def default-bucket    "witan-data")
+(def default-profile   "witan")
+(def default-directory "datasets/default_datasets")
+(def default-folder    "witan.models.demography")
+(def filename-regex    #".csv")
+
+(defn upload-file
+  [dir bucket file]
+  (when (and (not (.isDirectory file))
```

Comment:
I'd consider moving this when logic to a filter before upload-file is called. Cause at the moment this function is really upload-file-if-its-not-a-dir-but-is-a-csv-file :)